e2e/test/scenarios/sharing/alert/email-alert.cy.spec.js

```diff
@@ -64,25 +64,60 @@ describe("scenarios > alert > email_alert", { tags: "@external" }, () => {
       table: PEOPLE_ID,
     });
 
-    cy.icon("bell").click();
-
-    cy.findByRole("dialog").within(() => {
-      cy.findByLabelText("Name").type(" alert");
-      cy.findByRole("button", { name: "Save" }).click();
-    });
-
-    cy.wait("@saveCard");
+    saveAlert();
 
-    cy.findByTestId("alert-education-screen").within(() => {
-      cy.findByRole("button", { name: "Set up an alert" }).click();
+    cy.findByTestId("toast-undo").within(() => {
+      cy.findByText("Your alert is all set up.").should("be.visible");
     });
-    cy.findByRole("button", { name: "Done" }).click();
 
     cy.wait("@savedAlert").then(({ response: { body } }) => {
       expect(body.channels[0].channel_type).to.eq("email");
       expect(body.channels[0].enabled).to.eq(true);
     });
   });
+
+  it("should enable alert to be updated (without updating question) (metabase#36866)", () => {
+    openTable({
+      table: PEOPLE_ID,
+    });
+
+    saveAlert();
+
+    cy.log("Check that /api/card has been called once");
+    cy.get("@saveCard.all").should("have.length", 1);
+
+    cy.findByTestId("toast-undo").within(() => {
+      cy.findByText("Your alert is all set up.").should("be.visible");
+    });
+
+    clickAlertBell();
+
+    cy.findByTestId("popover").within(() => {
+      cy.findByText("You set up an alert").should("be.visible");
+      cy.findByText("Edit").click();
+    });
+
+    cy.log("Change the frequency of the alert to weekly");
+
+    cy.get(".Modal--full").within(() => {
```

Comment:
Another approach would be to add some attributes to [this full-page modal](https://github.com/metabase/metabase/blob/0634baa6200f17847d817ca38af0baa0e5ff4065/frontend/src/metabase/components/Modal/FullPageModal.tsx#L89), such as `role="dialog"` and `aria-labelledby="Edit your alert"` – or rather, the modal could have a `name` as a prop, and then we'd do `aria-labelledby={name}` – and then we could do `cy.findByRole('dialog', {name: "Edit your alert"})`, which is more user-centric. I'd be glad to do that, if reviewers prefer. Since that would involve adding a prop to a  For now, I've chosen a simpler option. I see the codebase already uses `cy.get('.Modal--full')` a few times, like [here](https://github.com/metabase/metabase/blob/0634baa6200f17847d817ca38af0baa0e5ff4065/e2e/test/scenarios/dashboard/dashboard.cy.spec.js#L877).

Related Slack thread: https://metaboat.slack.com/archives/C505ZNNH4/p1702557148449489?thread_ts=1702556878.526329&cid=C505ZNNH4

Reply:
good questions! 
- cy.get() is strongly discouraged. it exists in older tests, we shouldn't copy that pattern.
- accessible roles and other attributes that play nicely with react testing library are highly favored, but there are some existing helpers and tests that rely on a single "dialog" role being present, so if you're in a situation with multiple dialogs it could get messy.
- the easiest and safest option from a testing standpoint is using `data-testid`, it makes tests nice and easy to read because you can make it very specific, it's always safe to add, it's most likely to be stable, but it doesn't add any nice accessibility feature to the application.


TLDR be very bold:
![200w](https://github.com/metabase/metabase/assets/30528226/8953073c-796f-48b3-8d1c-62cc00c7053e)

Reply:
> accessible roles and other attributes that play nicely with react testing library are highly favored, but there are some existing helpers and tests that rely on a single "dialog" role being present, so if you're in a situation with multiple dialogs it could get messy.

I went ahead and tidied up some of the messiness around testing modals. The tidying happens in this new [refactoring PR](https://github.com/metabase/metabase/pull/37028). This "When updating an alert" PR now uses that refactoring PR as its base.